### ISSUE TITLE
[ml] supporting date_nanos as timestamp for the input

### ISSUE THREAD

**Issue by jguay** [elastic/elasticsearch#49889]
At present, trying to use a field of datatype `date_nanos` will return [error](https://github.com/elastic/kibana/blob/7.4/x-pack/legacy/plugins/ml/server/models/job_validation/messages.js#L378) : 
```
'fieldName' cannot be used as the time-field because it's not a valid field of type 'date'.
```

Some indices may only use timestamps with precision requiring datatype date_nanos, these indices cannot be used for input for machine learning jobs (unless another field of datatype date is added)

Note this feature request is limited to the input - it not related to being able to process data in nano seconds or below current second bucket_span raised in [644](https://github.com/elastic/ml-cpp/issues/644)

**Comment by davidkyle** [comment]
Using the ml APIs directly rather than kibana it is possible to create a run a job on data with a time field of type `date_nanos` I have tested this on version 7.5. 

The error mentioned above comes from the UI, I've raised elastic/kibana#52921 in place of this.

**Comment by sophiec20** [comment]
Re-opening. As it stands with `7.5` and prior, using timestamps containing data with precision of `data_nanos` is not yet supported and will be documented as a limitation. Both UI and back end work is required to add this functionality.

**Comment by droberts195** [comment]
> Using the ml APIs directly rather than kibana it is possible to create a run a job on data with a time field of type `date_nanos`

It turns out this was only true if the extra precision was not used.  In other words, a datafeed would work fine on the `date_nanos` type providing every timestamp was an exact number of milliseconds since the epoch, i.e. nanoseconds since the epoch was a multiple of 1000000.

#51180 fixes the case of a fractional number of milliseconds since the epoch.  (And note that the smallest possible ML `bucket_span` remains 1 second, and anomaly detection will still operate on summary statistics at the bucket level regardless of the underlying timestamp accuracy.)

**Comment by droberts195** [comment]
I believe this is fixed in 7.7 and above by https://github.com/elastic/kibana/pull/59017
